### PR TITLE
Adding burn::nn::Sigmoid

### PR DESCRIPTION
**Pull request by RuelYasa**
### Checklist

- [x] Confirmed that `run-checks all` script has been executed.
- [x] Made sure the book is up to date with changes in this PR.

### Related Issues/PRs

https://github.com/tracel-ai/burn/issues/2030

### Changes

Add a Sigmoid struct in nn just like Relu and Gelu etc.
Specifically:
1. Added sigmoid.rs in burn_core/src/nn
2. Added `mod sigmoid;` and `pub use sigmoid::*;` in burn_core/src/nn/mod.rs

### Testing

1. `cargo check` succeeded.
3. `mod tests` succeeded.
4. `run-checks all` script failed due to
```
error: struct `ThreadId` is never constructed
  --> crates/burn-common/src/stub.rs:99:12
   |
99 | pub struct ThreadId(core::num::NonZeroU64);
   |            ^^^^^^^^
   |
```
, but it is seemingly an existing problem, which is irrelative to my modification.
[An earlier PR](https://github.com/tracel-ai/burn/pull/1948) also mentioned the problem.

### CODE REVIEW

File: crates/burn-core/src/nn/sigmoid.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+use crate as burn;
+
+use crate::module::Module;
+use crate::tensor::backend::Backend;
+use crate::tensor::Tensor;
+
+/// Applies the rectified linear unit function element-wise
+/// See also [sigmoid](burn::tensor::activation::sigmoid)
+///
```
Comment (laggui) [inline crates/burn-core/src/nn/sigmoid.rs:9]: Should state: "Applies the logistic sigmoid function element-wise"

Also, we can remove the empty comment line `///`

**Review by laggui — CHANGES_REQUESTED** [review]
Thanks for resolving your own issue 😄 

LGTM! Just a minor change regarding the docstring.

**Comment by RuelYasa** [comment]
Oops... I forgot to *git push*!
It should be alright now.

**Review by laggui — APPROVED** [review]
Thanks!

**Review by nathanielsimard — APPROVED** [review]

**Pull request merged**